homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,178 @@
+"""Backup manager for the Backup integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import hashlib
+import json
+from pathlib import Path
+import tarfile
+from tempfile import TemporaryDirectory
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util import json as json_util
+from homeassistant.util.dt import now
+
+from .const import EXCLUDE_FROM_BACKUP, LOGGER, VERSION
+
+
+@dataclass
+class Backup:
+    """Backup class."""
+
+    slug: str
+    name: str
+    date: str
+    path: Path
+    size: float
+
+    def as_dict(self) -> dict:
+        """Return a dict representation of this backup."""
+        return {
+            "slug": self.slug,
+            "name": self.name,
+            "date": self.date,
+            "size": self.size,
+            "path": self.path.as_posix(),
+        }
+
+
+class BackupManager:
+    """Backup manager for the Backup integration."""
+
+    _backups: dict[str, Backup] | None = None
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the backup manager."""
+        self.hass = hass
+        self.backup_dir = Path(hass.config.path("backups"))
+        self.backing_up = False
+
+    @property
+    def backups(self) -> dict[str, Backup] | None:
+        """Return a dict of backups."""
+        return self._backups
+
+    def load_backups(self) -> None:
+        """Load data of stored backup files."""
+        backups = {}
+
+        def _read_backup_info(backup_path: Path) -> None:
+            with tarfile.open(backup_path, "r:") as backup_file:
```

Comment:
You should use stream like the old tape bands. This flood the memory because it keeps everything into memory just to read a file out of it and the same for creating the backup itself

Reply:
<https://github.com/home-assistant/supervisor/blob/main/supervisor/backups/backup.py#L230-L238>

Reply:
uhh, that is a bug, securetar handle it all as stream which is right

Reply:
Update: Yeah, we do that for the top tarfile and use the stream for all others because of the random access to files. So it's not nice but work.